exp/addons.md

```diff
@@ -0,0 +1,42 @@
+Addons provide a way to extend and customise images in Anbox Cloud. Once you have created addons, you can create [hooks](https://discourse.ubuntu.com/t/28555) for them that are triggered based on events in the container life cycle. You can create addons independently and later attach it to individual applications.
+
+To create or update an addon, you need a specific file structure for the directory containing your addon files. In the directory where you created your addon files, also create the following:
+
+- A file named `manifest.yaml`. See [Addon manifest](https://discourse.ubuntu.com/t/25293) to learn about valid keys in an addon manifest file.
+- A directory named `hooks`. This directory must contain at least one executable file with a valid hook name.
+
+Other files in the addon directory are bundled with the addon. They can be accessed in a hook by using the `$ADDON_DIR` [environment variable](https://discourse.ubuntu.com/t/hooks/28555#env-variables).
+
+For example:
+
+```bash
+cat "$ADDON_DIR"/public_key.pem >> ~/.ssh/authorized_keys
+```
+
+To create an addon, you must provide the Anbox Management Client (AMC) with either of the following:
+* The addon directory
+* A tarball containing the required addon file structure
+* A zip archive containing the required addon file structure
+
+## Best practices
+
+Addons must be created with careful consideration to not affect performance adversely. A good addon is light-weight and are targeted for the necessary applications. 
```

Comment:
```suggestion
Addons must be created with careful consideration to not affect performance adversely. A good addon is light-weight and is targeted for the necessary applications. 
```

Also, isn't the second sentence a bit redundant with 'Keep addons light' below?

Reply:
Yes, but it is intentional. The lead in sentence after "Best practices" is to set the context in the reader's mind while we elaborate more in the later sections. I don't feel strongly about removing it as well. Let me know what you think.

Reply:
I think it's fine if it's intentional, thanks for the answer!